iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/RegistryManagerTests.java

```diff
@@ -459,35 +459,38 @@ public void apply_configuration_e2e() throws Exception
     public void deviceCreationWithDeviceScope() throws IOException, InterruptedException, IotHubException, URISyntaxException
     {
         // Arrange
-        RegistryManagerTestInstance testInstance = new RegistryManagerTestInstance();
-        deleteDeviceIfItExistsAlready(testInstance.registryManager, testInstance.deviceId);
+        String edge1Id = deviceIdPrefix + UUID.randomUUID().toString();
+        String edge2Id = deviceIdPrefix + UUID.randomUUID().toString();
+        String deviceId = deviceIdPrefix + UUID.randomUUID().toString();
 
         //-Create-//
-        Device edgeDevice1 = Device.createFromId(testInstance.deviceId, DeviceStatus.Enabled, null);
+        Device edgeDevice1 = Device.createFromId(edge1Id, DeviceStatus.Enabled, null);
         DeviceCapabilities capabilities = new DeviceCapabilities();
         capabilities.setIotEdge(true);
         edgeDevice1.setCapabilities(capabilities);
-        edgeDevice1 = Tools.addDeviceWithRetry(testInstance.registryManager, edgeDevice1);
+        edgeDevice1 = Tools.addDeviceWithRetry(this.testInstance.registryManager, edgeDevice1);
 
-        Device edgeDevice2 = Device.createFromId(testInstance.deviceId, DeviceStatus.Enabled, null);
+        Device edgeDevice2 = Device.createFromId(edge2Id, DeviceStatus.Enabled, null);
         capabilities.setIotEdge(true);
         edgeDevice2.setCapabilities(capabilities);
         edgeDevice2.getParentScopes().add(edgeDevice1.getScope()); // set edge1 as parent
-        edgeDevice2 = Tools.addDeviceWithRetry(testInstance.registryManager, edgeDevice2);
+        edgeDevice2 = Tools.addDeviceWithRetry(this.testInstance.registryManager, edgeDevice2);
 
         Device leafDevice = Device.createFromId(
-                testInstance.deviceId + "-leaf",
+                this.testInstance.deviceId,
                 DeviceStatus.Enabled,
                 null);
         assertNotNull(edgeDevice1.getScope());
         leafDevice.setScope(edgeDevice1.getScope());
-        Tools.addDeviceWithRetry(testInstance.registryManager, leafDevice);
+        Tools.addDeviceWithRetry(this.testInstance.registryManager, leafDevice);
 
         //-Read-//
-        Device deviceRetrieved = testInstance.registryManager.getDevice(testInstance.deviceId);
+        Device deviceRetrieved = this.testInstance.registryManager.getDevice(deviceId);
 
         //-Delete-//
-        testInstance.registryManager.removeDevice(testInstance.deviceId);
+        this.testInstance.registryManager.removeDevice(edge1Id);
+        this.testInstance.registryManager.removeDevice(edge2Id);
+        this.testInstance.registryManager.removeDevice(this.testInstance.deviceId);
```

Comment:
Assuming this is same as deviceId won't need "this.testInstance."

Reply:
Good spot. I'll need to fix this too.